tests/ui/manual_ok_err.rs

```diff
@@ -0,0 +1,108 @@
+#![warn(clippy::manual_ok_err)]
+
+fn funcall() -> Result<u32, &'static str> {
+    todo!()
+}
+
+fn main() {
+    let _ = match funcall() {
```

Comment:
Indeed. I've added a check that the type is the same inside the `Result` variant we keep and the `Option`. I've also rebased this patch because I added a new function to `clippy_utils/src/ty/mod.rs` which has been renamed from `ty.rs` in the meantime.